core/API/Request.php

```diff
@@ -323,6 +323,10 @@ private static function forceReloadAuthUsingTokenAuth($tokenAuth)
 
     private static function shouldReloadAuthUsingTokenAuth($request)
     {
+        if (is_null($request)) {
```

Comment:
Problem is, that ```$request``` is ```null```. So the Auth was never reloaded.
I had searched for a better solution. But I didn't find another without changing the current logic.
There seems to be no way to determine whether a parameter is set in the request but empty. ```Common::getRequestVar()``` returns the default (or throws exception) if the parameter is missing or empty. Imho returning the empty value would be correct here, but I guess that might break other stuff in piwik to change that behavior.